src/Diagrams/Path.hs

```diff
@@ -90,12 +94,16 @@ import           Linear.Vector
 --   and they form a monoid under /superposition/ (placing one path on
 --   top of another) rather than concatenation.
 newtype Path v n = Path [Located (Trail v n)]
-  deriving (Semigroup, Monoid
+  deriving (Semigroup, Monoid, Generic
```

Comment:
We only support GHC >= 7.6.  [(travis)](https://travis-ci.org/diagrams/diagrams-lib) We'll probably stop supporting 7.6 when 8.0 comes out, and that's probably when we'll release these changes.  Thanks for asking.